pkg/controller/config/config_controller.go

```diff
@@ -254,6 +264,38 @@ func (r *ReconcileConfig) Reconcile(req reconcile.Request) (reconcile.Result, er
 		return reconcile.Result{}, err
 	}
 
+	namespaces := corev1.NamespaceList{}
+	err = r.client.List(context.TODO(), &namespaces, &client.ListOptions{})
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for _, namespace := range namespaces.Items {
+		if ignore, _ := regexp.MatchString(flag.IgnorePattern, namespace.Name); ignore {
+			continue
+		}
+
+		downgraded := false
+		for _, excludedNamespace := range cfg.Spec.NamespaceExclusions {
+			if excludedNamespace == "*" || excludedNamespace == namespace.Name {
+				err = r.downgradeDefaultSA(namespace)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				downgraded = true
+				break
+			}
+		}
+
+		if !downgraded {
+			err = r.upgradeDefaultSA(namespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	}
```

Comment:
shall we move this logic and sub function in the the `rbac controller` https://github.com/openshift/tektoncd-pipeline-operator/blob/74c0817d8c2ebaf98622e0994010c38696b4876d/pkg/controller/rbac/rbac_controller.go#L58

the rbac controller watches `namespaces` so it might be possible to achieve `downgradeSA/upgradeSA` without this `for loop`.

Reply:
I did consider that, but the event would be on the `Config` resource and not on the `namespace` 🤔

Reply:
Having said that the rbac controller too needs to check ..
1. when a new namespace shows up.
2. create SA with anyuid only if not on the exclusion list.